webui-src/app/chat/chat.js

```diff
@@ -108,69 +185,180 @@ const ChatLobbyModel = {
-        info
-      })),
-    ]),
-  };
-};
+const LobbyList = () => {
+    return {
+      view: vnode => {
+        rooms = vnode.attrs.rooms;
+        tagname = vnode.attrs.tagname;
+        lobbytagname= vnode.attrs.lobbytagname;
+        onclick=vnode.attrs.onclick||(() => null);
+        return [
+          m('hr'),
+          rooms.map(info => m(Lobby, {
+            info,
+            tagname: tagname,
+            lobbytagname,
+            onclick: onclick(info),
+          }))
+        ];
+      },
+    };
+}
+
+const SubscribedLeftLobbies = () => [
+  m('h5.lefttitle', 'subscribed:'),
+  m(LobbyList, {
+    rooms: sortLobbies(Object.values(ChatRoomsModel.subscribedRooms)),
+    tagname: '.leftlobby.subscribed',
+    lobbytagname:'h5.leftname',
+    onclick: ChatLobbyModel.switchToEvent,
+  })
+];
+
+const SubscribedLobbies = () => (
+  m('.widget', [
+    m('h3', 'Subscribed chat rooms'),
+    m(LobbyList, {
+      rooms: sortLobbies(Object.values(ChatRoomsModel.subscribedRooms)),
+      tagname:'.lobby.subscribed',
+      onclick: ChatLobbyModel.switchToEvent,
+    }),
+  ])
+);
+
+const PublicLeftLobbies = () => [
+  m('h5.lefttitle', 'public:'),
+  m(LobbyList, {
+    rooms: Object.values(ChatRoomsModel.allRooms).filter(info => !ChatRoomsModel.subscribed(info)),
+    tagname: '.leftlobby.public',
+    lobbytagname: 'h5.leftname',
+    onclick: ChatLobbyModel.setupEvent,
+  }),
+];
 
 const PublicLobbies = () => {
-  return {
-    oninit: () => ChatRoomsModel.loadPublicRooms(),
-    view: () => m('.widget', [
-      m('h3', 'Public chat rooms'),
-      m('hr'),
-      ChatRoomsModel.allRooms.filter(info => !ChatRoomsModel.subscribed(info)).map(info => m(Lobby, {
-            info,
-      })),
-    ])
-  };
+  return m('.widget', [
+    m('h3', 'Public chat rooms'),
+    m(LobbyList, {
+      rooms: ChatRoomsModel.allRooms.filter(info => !ChatRoomsModel.subscribed(info)),
+      tagname: '.lobby.public',
+      onclick: ChatLobbyModel.setupEvent,
+    }),
+  ])
 };
 
-const Layout = () => {
-  return {
-    view: vnode => m('.tab-page', [
-      m(SubscribedLobbies),
-      m(PublicLobbies),
-    ]),
-  };
+const LobbyName = () => {
+  return m('h3.lobbyName',
+    ChatLobbyModel.isSubscribed ? [
+      m('span.chatusername',ChatLobbyModel.lobby_user),
+      m('span.chatatchar','@'),
+    ] : [],
+    m('span.chatlobbyname',ChatLobbyModel.currentLobby.lobby_name),
+    m.route.param('subaction') !== 'setup' ? [
+      m('i.fas.fa-cog.setupicon', {
+        title: 'configure lobby',
+        onclick:() => m.route.set('/chat/:lobby/:subaction', {
+          lobby: m.route.param('lobby'),
+          subaction: 'setup'
+        }, true)
+      })
+    ] : [],
+    ChatLobbyModel.isSubscribed ? [
+      m('i.fas.fa-sign-out-alt.leaveicon', {
+        title: 'leaving lobby',
+        onclick:() => ChatLobbyModel.unsubscribeChatLobby(m.route.param('lobby'), () => {
+          m.route.set('/chat', null, true);
```

Comment:
Can simply do `m.route.set('/chat')`

Reply:
no, doesn't; will change semantics:

https://mithril.js.org/route.html#mrouteset 
 
Argument | Type | Required | Description
-- | -- | -- | --
options.replace | Boolean | No | Whether to create a new history entry or to replace the current one. Defaults to **false**

Reply:
I think you have to specify the option in an object
```suggestion
          m.route.set('/chat', null, { replace: true });
```